### PR TITLE
Add live time-period comparison to diff command

### PR DESCRIPTION
**Pull request by mivano**
## Summary

Adds the ability to compare two time periods directly by fetching data live from the Azure Cost API, without needing to manually export JSON files first.

### New Parameters
- `--source-from` (DateOnly) — start date for the source/baseline period
- `--source-to` (DateOnly) — end date for the source/baseline period
- The existing `--from`/`--to` params define the target (current) period

### Usage
```bash
# Live comparison (new)
azure-cost diff --source-from 2025-03-01 --source-to 2025-03-31 --from 2025-04-01 --to 2025-04-30

# File-based comparison (still works)
azure-cost diff --compare-to new.json --compare-from old.json
```

### Changes
- **DiffSettings.cs**: Added `SourceFrom`, `SourceTo` properties with `HasSourceDates`/`HasFileParams` helpers
- **DiffCommand.cs**: Injected `ICostRetriever` via constructor; added live data fetching path alongside existing file-based path; validates mutually exclusive modes
- **TextOutputFormatter.cs**: Implemented `WriteAccumulatedDiffCost` with comparison sections and summary
- **CsvOutputFormatter.cs**: Implemented `WriteAccumulatedDiffCost` with `DiffCsvRow` model
- **README.md**: Updated diff section with live comparison examples

### Tests
- 11 validation tests (source dates, file params, conflicts, edge cases)
- 8 formatter tests (Text and CSV output for diff)
- All 19 new tests pass

### CODE REVIEW

File: src/OutputFormatters/TextOutputFormatter.cs [hunk 1]
```diff
@@ -238,9 +238,63 @@ public override Task WriteAnomalyDetectionResults(DetectAnomalySettings settings
     public override Task WriteAccumulatedDiffCost(DiffSettings settings, AccumulatedCostDetails accumulatedCostSource,
         AccumulatedCostDetails accumulatedCostTarget)
     {
+        var currency = settings.UseUSD ? "USD" : accumulatedCostSource.Costs.FirstOrDefault()?.Currency ?? "USD";
+        
+        var sourceRange = $"{accumulatedCostSource.Costs.Min(a => a.Date)} to {accumulatedCostSource.Costs.Max(a => a.Date)}";
+        var targetRange = $"{accumulatedCostTarget.Costs.Min(a => a.Date)} to {accumulatedCostTarget.Costs.Max(a => a.Date)}";
+
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/TextOutputFormatter.cs]: Fixed in 721e83d. Added `.Any()` checks for both source and target costs, an early return with "No data found" when both are empty, fallback currency resolution from either side, and "N/A" ranges when a side has no data.

File: src/OutputFormatters/CsvOutputFormatter.cs [hunk 2]
```diff
@@ -149,6 +149,63 @@ public override Task WritePricesPerRegion(WhatIfSettings settings,
...
+        
+        foreach (var name in allNames)
+        {
+            var sourceCost = sourceItems.Where(a => a.ItemName == name)
+                .Sum(a => useUsd ? a.CostUsd : a.Cost);
+            var targetCost = targetItems.Where(a => a.ItemName == name)
+                .Sum(a => useUsd ? a.CostUsd : a.Cost);
+            var currency = useUsd ? "USD" :
+                sourceItems.FirstOrDefault(a => a.ItemName == name)?.Currency ??
+                targetItems.FirstOrDefault(a => a.ItemName == name)?.Currency ?? "USD";
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/CsvOutputFormatter.cs]: Fixed in 721e83d. Replaced the O(n²) per-name linear scans with dictionary-based lookups via GroupBy/ToDictionary, keeping AddDiffRows linear.

File: src/OutputFormatters/CsvOutputFormatter.cs [hunk 3]
```diff
@@ -201,4 +258,14 @@ public override string ConvertToString(object value, IWriterRow row, MemberMapDa
...
+}
+
+public class DiffCsvRow
+{
+    public string Category { get; set; }
+    public string Name { get; set; }
+    public double SourceCost { get; set; }
+    public double TargetCost { get; set; }
+    public double Diff { get; set; }
+    public string Currency { get; set; }
```
[the remark this answers is not in the document]
Comment (mivano) [reply src/OutputFormatters/CsvOutputFormatter.cs]: Fixed in 721e83d. Initialized Category, Name, and Currency properties to string.Empty to satisfy nullable reference type checks.
